ecdsa/src/gadgets/curve.rs

```diff
@@ -21,19 +21,68 @@ impl<C: Curve> AffinePointTarget<C> {
     }
 }
 
-impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
-    pub fn constant_affine_point<C: Curve>(
+pub trait CircuitBuilderCurve<F: RichField + Extendable<D>, const D: usize> {
```

Comment:
Hmm this feels a little hacky 🤔 But I feel like it's probably the best solution!

Reply:
Yeah definitely hacky haha. I think it's either that or changing all gadgets to take a `builder: &mut CircuitBuilder` which would be quite annoying.